util/cosmos.js

```diff
@@ -1,43 +1,29 @@
-import axios from 'axios';
-import { timeout } from '../common/util/misc';
+import { StargateClient } from '@cosmjs/stargate';
 
-const api = axios.create({
-  baseURL: '/api/proxy/cosmos/txs',
-  headers: {
-    'Content-Type': 'application/json',
-  },
-});
+import { COSMOS_ENDPOINT } from '../server/config/config';
+import { timeout } from '../common/util/misc';
 
-export async function waitForTxToBeMined(txHash) {
+export default async function waitForTxToBeMined(txHash) {
   let tx;
   let notFoundOnce = false;
+  const client = await StargateClient.connect(COSMOS_ENDPOINT);
   while (!tx) {
     /* eslint-disable no-await-in-loop */
     await timeout(1000);
     try {
-      const { data } = await api.get(`/${txHash}`);
-      if (data && data.height) {
+      const data = await client.getTx(txHash);
+      if (data) {
         ({ tx } = data);
-        const {
-          code,
-          logs: [{ success = false } = {}] = [],
-        } = data;
-        const isFailed = (code && code !== '0') || !success;
-        if (isFailed) throw new Error(code);
+        const { code } = data;
+        if (code) throw new Error(code);
```

Comment:
In stargate, `logs` is no longer available in `data`. So I check if `code` is not 0 only.
(See type `IndexedTx` diff in [launchpad](https://github.com/cosmos/cosmjs/blob/main/packages/launchpad/src/cosmosclient.ts#L104) & [stargate](https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/stargateclient.ts#L62))